### PR TITLE
add markdownlint-cli2

### LINKED ISSUE #26109: Package request: markdonwlint-cli2
### Package name

markdonwlint-cli2

### Package version

Newest

### Package website

https://github.com/DavidAnson/markdownlint-cli2

### Package availability

npm

### Additional comments

_No response_

### Package is not available

- [X] The package is not available on conda-forge.

### No previous issues or open PRs

- [X] No previous issue exists and no PR has been opened.

### PR DESCRIPTION
**Pull request by mahf708**
Fix #26109 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Review by xylar — APPROVED** [review]
@mahf708, this looks reasonable to me.

File: recipes/markdownlint-cli2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    tgz=$(npm pack)
+    npm install -g $tgz
+
+requirements:
+  build:
+    - nodejs
+
+test:
+  commands:
+    - npm list -g markdownlint-cli2
```
Comment (xhochy) [inline recipes/markdownlint-cli2/meta.yaml:25]: Shouldn't this expose a CLI you can query using `--help`?

**Review by xhochy — CHANGES_REQUESTED** [review]
Please have a look at the build script at https://github.com/conda-forge/pnpm-licenses-feedstock/blob/main/recipe/build.sh

**Comment by xylar** [comment]
@mahf708, he's presumably looking for you to change the testing to query the executable with `--help` instead of just making sure it exists.

File: recipes/markdownlint-cli2/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipes/markdownlint-cli2/meta.yaml:25]:
```suggestion
    - markdownlint-cli2 --help
```

**Review by xhochy — CHANGES_REQUESTED** [review]
Please use the script I referenced to ensure that binaries arw rebuilt and licenses are correctly bundled.

**Review by xhochy — APPROVED** [review]

**Comment by xhochy** [comment]
> Thanks @xhochy!!! One last question: is it accurate to have this be noarch: generic or should we build for linux/osx explicitly?

As all npm dependencies are also pure JS, it can stay noarch. CI on the feedstock will run only on Linux.

**Pull request merged**